backend/src/test/java/com/redhat/cloud/notifications/routers/DrawerResourceTest.java

```diff
@@ -0,0 +1,304 @@
+        createDrawerNotification(USERNAME2, secondEventInThePast);
+
+        Header defaultIdentityHeader = mockRbac(DEFAULT_ACCOUNT_ID, DEFAULT_ORG_ID, USERNAME, FULL_ACCESS);
+        Header identityHeaderUser2 = mockRbac(DEFAULT_ACCOUNT_ID, DEFAULT_ORG_ID, USERNAME2, FULL_ACCESS);
+
+        // should return 3 results
+        Page<DrawerEntryPayload> page = getDrawerEntries(defaultIdentityHeader, null, null, null, null, null, null, null, null, null, null);
+        assertEquals(3, page.getData().size());
+
+        // should return 3 results
+        page = getDrawerEntries(defaultIdentityHeader, Set.of(createdBundle2.getId(), createdBundle.getId()), null, null, null, null, null, null, null, null, null);
+        assertEquals(3, page.getData().size());
+
+        // should return 1 result
+        page = getDrawerEntries(defaultIdentityHeader, Set.of(createdBundle2.getId()), null, null, null, null, null, null, null, null, null);
+        assertEquals(1, page.getData().size());
+
+        // should return 3 results, bundle parameter should be ignored
+        page = getDrawerEntries(defaultIdentityHeader, Set.of(createdBundle2.getId()), Set.of(createdApplication.getId(), createdApplication2.getId()), null, null, null, null, null, null, null, null);
+        assertEquals(3, page.getData().size());
+
+        // should return 1 result
+        page = getDrawerEntries(defaultIdentityHeader, null, Set.of(createdApplication2.getId()), null, null, null, null, null, null, null, null);
+        assertEquals(1, page.getData().size());
+
+        // should return 3 results
+        page = getDrawerEntries(defaultIdentityHeader, null, null, Set.of(createdEventType.getId(), createdEventType2.getId(), createdEventType3.getId()), null, null, null, null, null, null, null);
+        assertEquals(3, page.getData().size());
+
+        // should return 1 result
+        page = getDrawerEntries(defaultIdentityHeader, null, null, Set.of(createdEventType3.getId()), null, null, null, null, null, null, null);
+        assertEquals(1, page.getData().size());
+
+        // should return 1 result, application parameter should be ignored
+        page = getDrawerEntries(defaultIdentityHeader, null, Set.of(createdApplication.getId()), Set.of(createdEventType3.getId()), null, null, null, null, null, null, null);
+        assertEquals(1, page.getData().size());
+
+        // should return 3 results
+        page = getDrawerEntries(identityHeaderUser2, null, null, null, null, LocalDateTime.now(UTC), null, null, null, null, null);
+        assertEquals(3, page.getData().size());
+
+        // should return 2 results
+        page = getDrawerEntries(identityHeaderUser2, null, null, null, null, LocalDateTime.now(UTC).minusDays(1), null, null, null, null, null);
+        assertEquals(2, page.getData().size());
```

Comment:
We'll need to improve these tests and run assertions on the returned data. The current tests are good enough for this PR so let's do that in a subsequent PR.